compiler/damlc/tests/daml-test-files/InterfaceConversions.daml

```diff
@@ -45,58 +46,99 @@ template Template2
+                x2 === None
+                x3 === None
+                x4 === Some (templateCid2, template2)
+
+        nonconsuming choice Call2 : ()
+            controller party
+            do useActionFrom party $ \_cid -> pure ()
+
+        nonconsuming choice Call3 : ()
+            controller party
+            do useActionFrom party $ \cid -> void (fetch cid)
+
+        nonconsuming choice Call4 : ()
+            controller party
+            do useActionFrom party $ \cid -> archive cid
+        
+        nonconsuming choice Call5 : ()
+            controller party
+            do useActionCoerce party $ \_cid -> pure ()
+
+        nonconsuming choice Call6 : ()
+            controller party
+            do useActionCoerce party $ \cid -> exercise cid MyChoice2
+
+        nonconsuming choice Call7 : ()
+            controller party
+            do useActionCoerce party $ \cid -> void (fetch cid)
+
+        nonconsuming choice Call8 : ()
+            controller party
+            do useActionCoerce party $ \cid -> archive cid
+
+main = script do
+    p <- allocateParty "Alice"
+
+    testCid <- p `submit` createCmd (Test p)
+
+    p `submit` exerciseCmd testCid Call1
 
     -- Test that using fromInterfaceContractId incorrectly will result
     -- in failed fetch/exercise/archives. I.e. using a Template1 contract
     -- id as if it were a Template2 contract id will always fail, even
     -- for actions that make sense for the common interface Iface.
-    let useAction : (ContractId Template2 -> Update ()) -> Update ()
-        useAction action = do
-            cid <- create (toInterface @Iface (Template1 p 10))
-            action (fromInterfaceContractId @Template2 cid)
-    submit p do useAction $ \_cid -> pure ()
-    submitMustFail p do useAction $ \cid -> void (fetch cid)
-    submitMustFail p do useAction $ \cid -> archive cid
+    -- See useActionFrom
+    p `submit` exerciseCmd testCid Call2
+    p `submitMustFail` exerciseCmd testCid Call3
+    p `submitMustFail` exerciseCmd testCid Call4
 
     -- Test that using coerceInterfaceContractId incorrectly will result
     -- in failed fetch/exercise/archives. I.e. using an Iface contract
     -- id as if it were a Iface2 contract id will always fail.
-    let useAction : (ContractId Iface2 -> Update ()) -> Update ()
-        useAction action = do
-            cid <- create (toInterface @Iface (Template1 p 10))
-            action (coerceInterfaceContractId @Iface2 cid)
-    submit p do useAction $ \_cid -> pure ()
-    submitMustFail p do useAction $ \cid -> exercise cid MyChoice2
-    submitMustFail p do useAction $ \cid -> void (fetch cid)
-    submitMustFail p do useAction $ \cid -> archive cid
-
--- @ENABLE-SCENARIOS
+    -- See useActionCoerce
+    p `submit` exerciseCmd testCid Call5
+    p `submitMustFail` exerciseCmd testCid Call6
+    p `submitMustFail` exerciseCmd testCid Call7
+    p `submitMustFail` exerciseCmd testCid Call8
```

Comment:
Not necessary for this ticket, but here it would be nice to have a script by test.